backend/src/apiserver/client/swf.go

```diff
@@ -19,21 +19,41 @@ import (
 
 	"github.com/cenkalti/backoff"
 	"github.com/golang/glog"
+	"github.com/kubeflow/pipelines/backend/src/apiserver/common"
 	swfclient "github.com/kubeflow/pipelines/backend/src/crd/pkg/client/clientset/versioned"
 	"github.com/kubeflow/pipelines/backend/src/crd/pkg/client/clientset/versioned/typed/scheduledworkflow/v1beta1"
 	"k8s.io/client-go/rest"
 )
 
+const (
+	PodNamespace = "POD_NAMESPACE"
+)
+
+type SwfClientInterface interface {
+	ScheduledWorkflow(namespace string) v1beta1.ScheduledWorkflowInterface
+}
+
+type SwfClient struct {
+	swfV1beta1Client v1beta1.ScheduledworkflowV1beta1Interface
+}
+
+func (swfClient *SwfClient) ScheduledWorkflow(namespace string) v1beta1.ScheduledWorkflowInterface {
+	if len(namespace) == 0 {
+		namespace = common.GetStringConfig(PodNamespace)
+	}
```

Comment:
discussion: IMO, we shouldn't read configuration as default value here, because it's impossible to figure out if the default value is reasonable here.
I can imagine a few types of error:
* User running single user mode
  * Forgot to set POD_NAMESPACE env in deployment, we should let user know POD_NAMESPACE is missing
* User running multi user mode
  * Forgot to pass namespace in api request, we should let user know namespace is required

With current implementation, we cannot distinguish above errors, especially in case `Forgot to pass namespace in api request`, user probably gets a runtime error when the workflow is run in kubeflow namespace and some resource is missing. That's really hard to troubleshoot.

Therefore, I think it's better to figure out why a parameter is missing with enough context and generate a meaningful error message.

It's also the pattern for argo client: https://github.com/kubeflow/pipelines/blame/3d8077cc37779efc42aa09439f7ac74e12cb5b4b/backend/src/apiserver/client/argo.go#L37

Reply:
I did a similar refactoring in https://github.com/kubeflow/pipelines/commit/dd96b4601d9cf4022569b91fc2c6b8c384675e03#diff-cec21bb0d4967636d39476b4cd34a0ce for similar reasons as above.

Reply:
After reading code below, I realized you have already added error handling for cases mentioned above. That's great!

Therefore, I think you don't need to default to POD_NAMESPACE here again, it should be handled at server/resource_manager level.

Reply:
Make sense. Updated.